libmdnsd/CMakeLists.txt

```diff
@@ -25,7 +27,7 @@ if(NOT MDNSD_COMPILE_AS_CXX AND (CMAKE_COMPILER_IS_GNUCC OR "x${CMAKE_C_COMPILER
     add_definitions(-Wpedantic)
   endif()
   if(NOT WIN32 AND NOT CYGWIN)
-    add_definitions(-Wshadow -Wconversion -fvisibility=hidden -fPIC)
```

Comment:
You can remove the `-fvisibility=hidden`,
by doing `set(CMAKE_C_VISIBILITY_PRESET hidden)`
You probably also need `set(CMAKE_CXX_VISIBILITY_PRESET hidden)` when you compile the sources as `CXX`.
See https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_VISIBILITY_PRESET.html

Reply:
This can be done outside a `if`.